File: src/main/java/dev/swe573/whatsthis/dto/UserDto.java
package dev.swe573.whatsthis.dto;

import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;
import java.util.Set;

@Data
public class UserDto {

    private Long id;
    private String username;
    private String email;
    private String password; //This is going to change. It is here for testing.
    private Set<String> roles;
    private String bio;
    private String profilePictureUrl;
    private LocalDateTime createdAt;
    private List<Long> commentIds;

}

File: mobile/eurekaNexus/app/src/main/java/com/yusuf/eurekanexus/data/model/UserDto.kt
package com.yusuf.eurekanexus.data.model

import com.google.gson.annotations.SerializedName

data class UserDto(
    @SerializedName("username") val username: String,
    @SerializedName("email") val email: String,
    @SerializedName("password") val password: String,
    @SerializedName("roles") val roles: List<String>,
    @SerializedName("bio") val bio: String?,
    @SerializedName("profilePictureUrl") val profilePictureUrl: String?,
    @SerializedName("createdAt") val createdAt: String?,
    @SerializedName("commentIds") val commentIds: List<Long>?
)